src/otx/v2/adapters/torch/mmengine/mmaction/modules/datasets/pipelines/loading.py

```diff
@@ -0,0 +1,49 @@
+"""Collection of video loading data pipelines for OTX Action Task."""
+
+# Copyright (C) 2022 Intel Corporation
+# SPDX-License-Identifier: Apache-2.0
+
+from __future__ import annotations
+
+from typing import TYPE_CHECKING, Any
+
+import numpy as np
+from mmaction.registry import TRANSFORMS
+
+if TYPE_CHECKING:
+    from datumaro.components.dataset import Dataset as DatumDataset
+
+@TRANSFORMS.register_module(force=True)
+class OTXRawFrameDecode:
+    """Load and decode frames with given indices."""
+
+    otx_dataset: DatumDataset
+
+    def __call__(self, results: dict[str, Any]) -> dict[str, Any]:
+        """Call function of RawFrameDecode."""
+        return self._decode_from_list(results)
+
+    def _decode_from_list(self, results: dict[str, Any]) -> dict:
+        """Generate numpy array list from list of DatasetItemEntity."""
+        imgs = []
+        for index in results["frame_inds"]:
+            item_id = results["item_ids"][index]
+            imgs.append(self.otx_dataset.get(id=item_id, subset="annotations").media.data)
```

Comment:
`subset="annotations"` seems like hard-coding. Are we able to have `subset` info in `results`?

Reply:
In detection case, subset is "train" and "val". Why action has "annotation" subset?

Reply:
This comes from the Datumaro importer. CVAT importer used the folder name to set the subset name. and for our case, CVAT format used `annotations` and `images` folder.

<img width="384" alt="image" src="https://github.com/openvinotoolkit/training_extensions/assets/80735344/0842e5e2-1546-4454-8579-a13729e1144c">